modules/entity_version_history/tests/src/FunctionalJavascript/HistoryTabSettingsTest.php

```diff
@@ -0,0 +1,169 @@
+use Drupal\Tests\SchemaCheckTestTrait;
+
+/**
+ * Ensures the entity version history config is correctly saved.
+ *
+ * @group entity_version_history
+ */
+class HistoryTabSettingsTest extends WebDriverTestBase {
+
+  use SchemaCheckTestTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'node',
+    'entity_version',
+    'entity_version_history',
+    'entity_version_history_test',
+  ];
+
+  /**
+   * A user with administrative permissions.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $adminUser;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Create user.
+    $this->adminUser = $this->drupalCreateUser([
+      'access entity version history configuration',
+      'access administration pages',
+    ]);
+    $this->drupalLogin($this->adminUser);
+  }
+
+  /**
+   * Tests whether the history settings form is correctly saving the settings.
+   */
+  public function testHistoryTabSettingsForm() {
+    $this->drupalGet('admin/config/entity-version/history-tab');
+    $page = $this->getSession()->getPage();
+    $assert_session = $this->assertSession();
+
+    // Check we have the entity type checkboxes.
+    $node_entity_checkbox = $assert_session->elementExists('css', '#edit-entity-types-node');
+    $history_entity_checkbox = $assert_session->elementExists('css', '#edit-entity-types-history-entity-test');
+
+    // Collect the bundle checkboxes and check that they are not visible.
+    $first_bundle_checkbox = $assert_session->elementExists('css', '#edit-node-first-bundle');
+    $second_bundle_checkbox = $assert_session->elementExists('css', '#edit-node-second-bundle');
+    $history_bundle_checkbox = $assert_session->elementExists('css', '#edit-history-entity-test-history-entity-test');
+
+    $this->assertFalse($first_bundle_checkbox->isVisible());
+    $this->assertFalse($second_bundle_checkbox->isVisible());
+    $this->assertFalse($history_bundle_checkbox->isVisible());
+
+    // Check the entity type checkboxes.
+    $node_entity_checkbox->check();
+    $history_entity_checkbox->check();
+
+    // Check the bundle checkboxes are now visible.
+    $this->assertTrue($first_bundle_checkbox->isVisible());
+    $this->assertTrue($second_bundle_checkbox->isVisible());
+    $this->assertTrue($history_bundle_checkbox->isVisible());
+
+    // Check that there is only one select field and it's not visible.
+    $selects = $page->findAll('css', 'details select');
+    $this->assertEquals(1, count($selects));
+    $select = reset($selects);
+    $this->assertEquals('node_second_bundle', $select->getAttribute('name'));
+    $this->assertFalse($select->isVisible());
+
+    // Check the bundle checkboxes.
+    $first_bundle_checkbox->check();
+    $second_bundle_checkbox->check();
+    $history_bundle_checkbox->check();
+
+    // Check that the select field has appeared.
+    $this->assertTrue($select->isVisible());
```

Comment:
No assertions on the options of the Field form element? One is not accessible but one should have two options. Also, the lack of this test shows that your form is broken. I added 2 fields on a bundle, changed from one to another but the form still shows me the old value, even if the config was saved correctly.

Reply:
That could happen if the machine name of the field is the same.

Reply:
Can you tell me how to reproduce the issue?